docs: add getting started guide for argocd-agent using kind

bianbbc87 opened the pull request:
**What does this PR do / why we need it**:
This PR adds documentation for setting up a local development environment for argocd-agent using kind.
The guide enables local multi-cluster configuration on macOS + kind without installing any additional packages—only relying on NodePort networking.

One of the major issues with kind environments is that, although clusters share the same Docker network, they use identical CIDR ranges, which inevitably causes Pod-to-Pod communication conflicts.
To resolve this, the guide explicitly defines separate CIDR ranges when creating clusters:

```bash
# principal
PRINCIPAL_POD_CIDR, PRINCIPAL_SVC_CIDR
# agent
AGENT_POD_CIDR, AGENT_SVC_CIDR
```
If you want to add another agent, simply increase the AGENT_USER_ID value by +1.

**Which issue(s) this PR fixes**:

Fixes #409 

**How to test changes / Special notes to the reviewer**:
How to Test:
Before running the test, I applied [PR #618](https://github.com/argoproj-labs/argocd-agent/pull/618), which resolves Redis egress and ingress issues.
The guide is designed so that you can simply follow it step by step using copy and paste.
If any issues occur during setup, please refer to the three troubleshooting cases documented at the bottom of the guide.

Special notes to the reviewer:
The original guide (kubernetes/index.md) instructed to run `2. pki init -> 3. principal install`, but this caused issues because principal install had not been executed beforehand.
Therefore, in this document, the steps are performed in the order `2. principal install -> 3. pki init`.

This document has been separated into a standalone file because it has not yet been fully integrated into the existing documentation flow.
I’d like to hear your thoughts on how this document could be best integrated.

Additionally, I would appreciate feedback on whether setting up environment variables at the beginning of the document improves usability.

Suggestions and feedback are also especially welcome.
Thank you! 🙏

**Checklist**

* [x] Documentation update is required by this PR (and has been updated) OR no documentation update is required.

bianbbc87 commented:
<img width="1362" height="577" alt="image" src="https://github.com/user-attachments/assets/487c1949-c7b2-4cf9-af23-e766efa70d1d" />

bianbbc87 commented on docs/getting-started/kubernetes/kind-index.md:
```diff
@@ -0,0 +1,698 @@
+# Getting Started
+
+## Table of Contents
```
Do you think this document should be integrated into the existing guide?
Personally, I think it’s helpful to include export environment variables directly in the guide, since kind requires managing different Pod and Service CIDRs across clusters.

I’d love to hear your thoughts on this.

mikeshng commented:
> Yes, I’ve tested this document more than three times using the final version to make sure everything works properly all tests were successful.

@bianbbc87 Perhaps I didn't explain it clearly earlier and my apologies for that. 
When I said “_have you built this doc locally and viewed it in your local browser_.” to be explicit, I meant have you ran the following commands so you can view the doc in your browser:
```
$ make build-docs
...
INFO    -  Documentation built in 0.72 seconds
```
and/or
```
$ make serve-docs 
...
INFO    -  [01:53:20] Serving on http://0.0.0.0:8000/
```

Please make sure everything runs ok locally first. Thanks again for your contribution!

mikeshng commented:
> If needed, I’d be happy to create another PR to check and address them.

Another PR would be great :) Thanks!

mikeshng commented on docs/getting-started/kubernetes/kind/index.md:
```diff
@@ -0,0 +1,619 @@
+# Getting Started with Kind
+
```
You should give an overview of what this doc is about similar to the other quick start guides. ".... setup argocd-agent using `kind` with a link to the kind website.

mikeshng commented on docs/getting-started/kubernetes/kind/index.md:
````diff
@@ -0,0 +1,619 @@
+export PATH=$PATH:$(pwd)/dist
+```
+
+## Architecture Overview
````
Can you make this diagram more Kind relevant like you gave the Kind clusters some names so maybe they should be marked in the diagram as well?

mikeshng reviewed (commented):
Some nits / suggestions. I will provide more feedback tomorrow.

bianbbc87 commented on docs/getting-started/kubernetes/kind/index.md:
```diff
@@ -0,0 +1,619 @@
+- `kubectl` (v1.20 or later)
+- `argocd-agentctl` CLI tool ([download from releases](https://github.com/argoproj-labs/argocd-agent/releases)) or you can install locally based on this document
+- `kustomize` (optional, for customization)
+- `kind` you can install locally based on this document
```
Thanks! Updated as suggested removed the "locally" mentions for both `argocd-agentctl` and `kind`, and added a link to the official Kind installation guide.

```diff
- `kubectl` (v1.20 or later)
- - `argocd-agentctl` CLI tool ([download from releases](https://github.com/argoproj-labs/argocd-agent/releases)) or you can install locally based on this document
+ - `argocd-agentctl` CLI tool ([download from releases](https://github.com/argoproj-labs/argocd-agent/releases))
- `kustomize` (optional, for customization)
- - `kind` you can install locally based on this document
+ - `kind` ([Kind Installation guide](https://kind.sigs.k8s.io/docs/user/quick-start/#installation))

bianbbc87 commented on docs/getting-started/kubernetes/kind/index.md:
````diff
@@ -0,0 +1,619 @@
+brew install kind
+```
+
+### Clone argocd-agent repository and build CLI
````
yes, this part overlaps with the _Prerequisites_ section above.  
I originally included it to keep the option open for users who may want to build `argocd-agentctl` locally,  
especially for those interested in contributing to the CLI itself.

Is there a separate document that covers the local build process for `argocd-agentctl`?  
If so, I'd like to link to it instead.

For now, I've removed the _Clone argocd-agent repository and build CLI_ section from the code since it’s redundant.

mikeshng replied:
I don't think we have a dev guide section yet and building the CLI will fall under that. This doc in my opinion should focus on quickly setup the argocd-agent env locally using KinD. It's kind of weird to have the instruction for building the CLI but not the argocd-agent image itself.

bianbbc87 replied:
To be precise, I’d like to add a new document specifically for developers.
I believe this document should be placed in a separate path rather than under Getting Started.

The content will be the same as the previous version of this commit, which used ./dist/ or install/.

mikeshng replied:
LGTM. Great job on this doc once again! Since it is a brand new section rather than just small fixes, I will leave it to Jann to decide when it's the best time to merge.

bianbbc87 commented:
I've also made a few small updates. @mikeshng 

1. Renamed **Cluster Name** -> **Set Environment Name** for better clarity  
3. Added a new section: **Adding Another Agent Cluster**

```diff
- ### Cluster Name
+ ### Set Environment Values

+ ## Adding Another Agent Cluster
+ To add another agent, simply **increment** the `AGENT_USER_ID` by `+1` and proceed to the **Workload Cluster Setup** section below.  
+ Each new agent automatically receives a **unique CIDR range** derived from its `AGENT_USER_ID` value.
+ 
+ ```bash
+ export AGENT_USER_ID=$((AGENT_USER_ID + 1))
+ export AGENT_POD_CIDR="10.$((244 + $AGENT_USER_ID)).0.0/16"
+ export AGENT_SVC_CIDR="10.$((96 + $AGENT_USER_ID)).0.0/12"
+ 
+ # (optional) Check variables
+ echo "Agent Pod CIDR: $AGENT_POD_CIDR"
+ echo "Agent Service CIDR: $AGENT_SVC_CIDR"
+ ```
+ 
+ !!! info "Next step"
+     Once the new agent variables are set, continue to the [Workload Cluster Setup](#workload-cluster-setup) section to create and configure the new agent cluster.

mikeshng commented on docs/getting-started/kubernetes/kind/index.md:
```diff
@@ -0,0 +1,627 @@
+kubectl get configmap argocd-agent-params \
+  -n "$NAMESPACE_NAME" \
+  --context "kind-$PRINCIPAL_CLUSTER_NAME" \
+  -o yaml | grep principal.allowed-namespaces
```
In other places you have `# Expected output` but not here.

bianbbc87 replied:
i added expected output in here
[d36f7dd](https://github.com/argoproj-labs/argocd-agent/pull/614/commits/d36f7dd33a9dca8204a811caf78a32c887afb2f6)

bianbbc87 commented on docs/getting-started/kubernetes/kind/index.md:
````diff
@@ -0,0 +1,627 @@
+
+```bash
+# Delete Application in wrong namespace
+kubectl delete application test-app -n $AGENT_APP_NAME --context kind-$PRINCIPAL_CLUSTER_NAME
````
That was a mistake!
I’ll make it consistent with the application name used during creation.

josephaw1022 commented:
good job! @bianbbc87
